src/covid19model/models/discrete_time.py

```diff
@@ -0,0 +1,294 @@
+import inspect
+import itertools
+
+import numpy as np
+from scipy.integrate import solve_ivp
+import xarray
+from scipy import interpolate as inter
+
+class DiscreteTimeModel:
+
+    state_names = None
+    parameter_names = None
+    parameters_stratified_names = None
+    stratification = None
+
+    def __init__(self, states, parameters):
```

Comment:
General comment here: we should try to avoid copy pasting large chunks of identical code, as that makes it a lot harder to maintain. Rather, we should see how we can share this base code for the model implementation. Using inheritance (making a "DiscreteBaseModel" subclass of "BaseModel"), that should be possible.


@twallema do you know what exactly is different here compared to the code in `base.py` `BaseModel` ?

Of course the `_sim_single` is different, but did you change things in other functions as well?

Reply:
Only two things are different: 1) the step size l is an optional argument of the sim function and is passed on to the discrete solver. 2) output is interpolated before assignment to the xarray in order to correspond with integer timesteps (makes handling easier when trying to match with data).